clients/src/main/java/org/apache/kafka/clients/ClientRequest.java

```diff
@@ -12,44 +12,63 @@
  */
 package org.apache.kafka.clients;
 
-import org.apache.kafka.common.requests.RequestSend;
+import org.apache.kafka.common.network.Send;
+import org.apache.kafka.common.requests.AbstractRequest;
+import org.apache.kafka.common.requests.RequestHeader;
 
 /**
  * A request being sent to the server. This holds both the network send as well as the client-level metadata.
  */
 public final class ClientRequest {
 
+    private final RequestHeader header;
+    private final AbstractRequest body;
     private final long createdTimeMs;
     private final boolean expectResponse;
-    private final RequestSend request;
+    private final Send send;
     private final RequestCompletionHandler callback;
     private final boolean isInitiatedByNetworkClient;
     private long sendTimeMs;
 
     /**
      * @param createdTimeMs The unix timestamp in milliseconds for the time at which this request was created.
      * @param expectResponse Should we expect a response message or is this request complete once it is sent?
-     * @param request The request
+     * @param header The request's header
+     * @param body The request's body
+     * @param send The send associated with the request
      * @param callback A callback to execute when the response has been received (or null if no callback is necessary)
      */
-    public ClientRequest(long createdTimeMs, boolean expectResponse, RequestSend request,
+    public ClientRequest(long createdTimeMs,
+                         boolean expectResponse,
+                         RequestHeader header,
+                         AbstractRequest body,
```

Comment:
Do we need body in ClientRequest? It seems that all usage of body is for testing.